src/quo2/components/tags/scrollable_view.cljs

```diff
@@ -0,0 +1,146 @@
+(ns quo2.components.tags.scrollable-view
+  (:require [reagent.core :as reagent]
+            [quo.react-native :as rn]
+            [oops.core :refer [oget]]
+            [status-im.ui.components.react :as react]
+            [status-im.utils.core :as utils]
+            [quo2.components.tabs.tab :as tab]
+            [quo2.components.tags.tag :as tag]
+            [status-im.utils.number :as number-utils]))
+
+(def default-tab-size 32)
+
+(defn calculate-fade-end-percentage
+  [{:keys [offset-x content-width layout-width max-fade-percentage]}]
+  (let [fade-percentage (max max-fade-percentage
+                             (/ (+ layout-width offset-x)
+                                content-width))]
+    ;; Truncate to avoid unnecessary rendering.
+    (if (> fade-percentage 0.99)
+      0.99
+      (number-utils/naive-round fade-percentage 2))))
+
+(defn view
+  "Usage:
+   {:type             :icon/:emoji/:label
+    :component        tag/tab
+    :size             32/24
+    :on-press         fn
+    :blurred?         true/false 
+    :labelled?        true/false
+    :disabled?        true/false
+    :scroll-on-press? true
+    :fade-end?        true
+    :on-change        fn
+    :default-active   tag-id
+    :data             [{:id :label \"\" :resource \"url\"}
+                       {:id :label \"\" :resource \"url\"}]}
+  Opts:
+   - `component` this is to determine which component is to be rendered since the 
+                 logic in this view is shared between tab and tag component
+   - `blurred`   boolean: use to determine border color if the background is blurred
+   - `type`      can be icon or emoji with or without a tag label
+   - `labelled`  boolean: is true if tag has label else false
+   - `size` number
+   - `scroll-on-press?` When non-nil, clicking on a tag centers it the middle
+  (with animation enabled).
+   - `fade-end?` When non-nil, causes the end of the scrollable view to fade out.
+   - `fade-end-percentage` Percentage where fading starts relative to the total
+  layout width of the `flat-list` data."
+
+  [{:keys [default-active fade-end-percentage]
+    :or   {fade-end-percentage 0.8}}]
+  (let [active-tab-id (reagent/atom default-active)
+        fading        (reagent/atom {:fade-end-percentage fade-end-percentage})
+        flat-list-ref (atom nil)]
+    (fn [{:keys [data fade-end-percentage fade-end? on-change on-scroll
+                 scroll-event-throttle scroll-on-press? size type labelled?
+                 disabled? blurred? icon-color blur? component
+                 override-theme]
+          :or   {fade-end-percentage fade-end-percentage fade-end? false
```

Comment:
[Code style] I find the line breaks here in order to save LoC, especially in the `:or` key, make the code rather difficult to read. Here's my suggestion if you like it:

```clojure
{fade-end-percentage   fade-end-percentage
 fade-end?             false
 scroll-event-throttle 64
 scroll-on-press?      false
 size                  default-tab-size}
```